provider/pkg/gen/schema.go

```diff
@@ -160,11 +172,21 @@ func PulumiSchema(swagger map[string]interface{}) pschema.PackageSpec {
 				for _, p := range kind.Properties() {
 					// JSONSchema type includes `$ref` and `$schema` properties, and $ is an invalid character in
 					// the generated names.
+					// TODO: this replacement is going to be deleted once pulumi/pulumi codegen supports masking $-fields.
```

Comment:
The final state is that the schema would contain fields named `$ref` and `$schema` and each language SDK needs to remove `$` appropriately.